app/services/message_employee_matcher.rb

```diff
@@ -4,7 +4,7 @@ def initialize(message)
   end
 
   def run
-    Employee.where(started_on: day_count.days.ago)
+    Employee.where(started_on: day_count.days.ago).select { |employee| match_time(employee.time_zone) && !message_already_sent?(employee) }
```

Comment:
there's a lot going on / to understand on this line - what about putting `Employee.where(started_on: day_count.days.ago).select { |employee| match_time(employee.time_zone)` into a small well-named method (similar to what you did with `message_already_sent?`) so it is easier to read and understand in smaller parts?